Mathlib/AlgebraicGeometry/PrimeSpectrum/TensorProduct.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.SurjectiveOnStalks
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+
+/-!
+
+# Lemmas regarding the prime spectrum of tensor products
+
+## Main result
+- `PrimeSpectrum.embedding_tensorProductTo_of_surjectiveOnStalks`:
+  If `R →+* T` is surjective on stalks,
+  then `Spec(S ⊗[R] T) → Spec S × Spec T` is a topological embedding
+  (where `Spec S × Spec T` is the cartesian product with the product topology).
+-/
+
+variable (R S T) [CommRing R] [CommRing S] [Algebra R S]
+variable [CommRing T] [Algebra R T] [Algebra S T] [IsScalarTower R S T]
+
+open TensorProduct
+
+/-- The canonical map from `Spec(S ⊗[R] T)` to the cartesian product `Spec S × Spec T`. -/
+noncomputable
+def PrimeSpectrum.tensorProductTo (x : PrimeSpectrum (S ⊗[R] T)) :
+    PrimeSpectrum S × PrimeSpectrum T :=
+  ⟨comap (algebraMap _ _) x, comap Algebra.TensorProduct.includeRight.toRingHom x⟩
+
+lemma PrimeSpectrum.continuous_tensorProductTo : Continuous (tensorProductTo R S T) :=
+  (comap _).2.prod_mk (comap _).2
```

Comment:
Maybe this should just be term of type `C(_, _)`?

Reply:
I think I prefer unbundled morphisms more. Plus the main point of this function is that it is a topological embedding (under assumptions) and not merely continuous functions.